src/compiler/checker.ts

```diff
@@ -16104,6 +16127,9 @@ namespace ts {
                         else if (deviation & ModifierFlags.Abstract) {
                             error(o.name, Diagnostics.Overload_signatures_must_all_be_abstract_or_non_abstract);
                         }
+                        else if (deviation & ModifierFlags.Override) {

```

Comment:
Remember to modify `const flagsToCheck` below to include `override`. And make a test for this.